src/test/java/org/reaktivity/ry/internal/RyMainTest.java

```diff
@@ -15,26 +15,106 @@
  */
 package org.reaktivity.ry.internal;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.reaktivity.ry.internal.RyTestCommandSpi.TEST_ARGUMENT;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+
+import javax.naming.spi.Resolver;
+
+import org.apache.ivy.Ivy;
+import org.apache.ivy.ant.AntWorkspaceResolver;
+import org.apache.ivy.ant.AntWorkspaceResolver.WorkspaceArtifact;
+import org.apache.ivy.core.install.InstallOptions;
+import org.apache.ivy.core.module.id.ModuleRevisionId;
+import org.apache.ivy.core.report.ResolveReport;
+import org.apache.ivy.core.settings.IvySettings;
+import org.apache.ivy.plugins.resolver.FileSystemResolver;
+import org.apache.ivy.util.DefaultMessageLogger;
+import org.apache.ivy.util.Message;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.reaktivity.ry.RyCommandSpi;
 
 public class RyMainTest
 {
+    @BeforeAll
+    static void setup()
+    {
+        DefaultMessageLogger logger = new DefaultMessageLogger(Message.MSG_VERBOSE);
+        Message.setDefaultLogger(logger);
+
+        // TODO Rename this variable, artifact name, and what version to use?
+        ModuleRevisionId reference = ModuleRevisionId.newInstance("reaktivity.org", "spi-test", "0.1");
+
+        InstallOptions options = new InstallOptions();
+
+        final String jarPath = "/tmp/a.jar"; // TODO Do something with this
+
+        try (
+            // TODO Fix these variable names
+            FileOutputStream fos = new FileOutputStream(jarPath);
+            JarOutputStream jos = new JarOutputStream(fos);
+            BufferedOutputStream bos = new BufferedOutputStream(jos);)
+        {
+            jos.putNextEntry(new JarEntry(String.format("/META-INF/services/%s", RyCommandSpi.class.getName())));
+            bos.write(RyTestCommandSpi.class.getName().getBytes());
+
+            AntWorkspaceResolver from = new AntWorkspaceResolver();
+            from.setName("workspace"); // TODO Make a constant
+            WorkspaceArtifact jarArtifact = from.createArtifact();
+            jarArtifact.setName(RyCommandSpi.class.getName());
+            jarArtifact.setPath(jarPath);
+            jarArtifact.setType("jar"); // TODO Defaults, so not necessary?
+            jarArtifact.setExt("jar"); // TODO Defaults, so not necessary?
+
+            FileSystemResolver localTest = new FileSystemResolver();
+            localTest.setName("local-test");
+            localTest.setM2compatible(true);
+            localTest.setLocal(true);
```

Comment:
Please call this `cache` both as variable name and resolver name.
No need to give variable names `Test` suffix, as it is clear we are already in a test method.